lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,825 @@
+        let currDur = nextTimeStamp - currTimeStamp;
+        let durCalcut = (nextTimeStamp-currTimeStamp)/1000;
+        // if(bolus[i].deviceTime > '2020-07-21 12:00:00' && bolus[i].deviceTime < '2020-07-21 14:00:00'){
+        //   debug('time',0)
+        // }
+        let boluMount = (parseInt(bolus[i].BolusRate) * 0.00625 * durCalcut) / (60 * 60);
+        if(bolus[i].BolusRate != '0' && parseInt(bolus[i].BolusRate) > 12800){
+          normal = boluMount + normal;
+          deviceTime = bolus[i].deviceTime;
+          index = bolus[i].recordId;
+        }
+        if(bolus[i].BolusRate != '0' && parseInt(bolus[i].BolusRate) < 12800){
+          square = boluMount+square;
+          dur = currDur+dur;
+        }
+      }
+      let postbolus = cfg.builder.makeDualBolus()
+      .with_normal(this.buildValue(normal))
+      .with_deviceTime(sundial.formatDeviceTime(new Date(deviceTime).valueOf()))
+      .with_extended(this.buildValue(square))
+      .with_duration(dur)
+      .set('index', index);
+      cfg.tzoUtil.fillInUTCInfo(postbolus, new Date(deviceTime).valueOf());
+      postbolus = postbolus.done();
+      postbolus.deviceId = 'MTM-I';
+      bolusArray.push(postbolus);
+      return bolusArray;
+    },
+
+    buildValue(originValue){
+      let value = this.formatDecimal(originValue,2);
+      let res =((value*1000)/25);
+      // console.log(value);
+      let floorRes = Math.floor(res);
+      let floor = floorRes*25;
+      if(res > floorRes) {
+        floor = (floor + 25)/1000;
+      }else{
+        floor = floor/1000;
+      }
+      return floor;
+    },
+    formatDecimal(originnum, decimal) {
+      let num = originnum.toString();
+      let index = num.indexOf('.');
+      let resNum = '';
+      if (index !== -1) {
+        resNum = num.substring(0, decimal + index + 1);
+      } else {
+        resNum = num.substring(0);
+      }
+      return parseFloat(resNum).toFixed(decimal);
+    },
+    fetchData(progress, data, cb) {
+      // debug('in fetchData', data);
+      let records = [];
+      let incomingA = data.incomingA;
+      const inComeRes = WeitaiUSB.parsePacket(incomingA,cfg,cb);  //inComeRes:{BloodGlucoses:[],BasalRates:[],BolusRates:[]}
+      data.BloodGlucoses = inComeRes.BloodGlucoses;
+      data.BasalRates = inComeRes.BasalRates;
+      data.BolusRates = inComeRes.BolusRates;
+      data.records = records;
+      return cb(null, data);
+    },
+
+    processData(progress, data, cb) {
+      progress(100);
+      let bloodRes = this.buildBlood(data.BloodGlucoses);
+      let basalRes = this.buildBasal(data.BasalRates);
+      // let basalRes = this.buildBasal(testBasal);
+      let bolusRes = this.buildBolus(data.BolusRates);
+      // let bolusRes = this.buildBolus(testArray);
+      basalRes = basalRes.length > 1 ? basalRes:[];
+      let post_records = [].concat(bloodRes,basalRes,bolusRes);
+      if(!post_records.length){
+        return cb({code:'E_WEITAI_NO_UPLOAD'}, data);
```

Comment:
```suggestion
        return cb(new Error('No new records to upload'), null);
```
Error handling in the driver itself should be done this way, where you create a new Error object for the first parameter, and don't send anything in the second parameter.